### DISCUSSION TITLE
Spotify social identity provider

### DISCUSSION THREAD

**Discussion by JiriHartikka** [keycloak/keycloak#8555] [Archive]
I am implementing a social identity provider for Spotify as part of a hobby project I am working on. I am wondering if this is something that I should try to get merged to the upstream repository.

Spotify does not support openid connect, and as I want my Keycloak users to be able to login using Spotify, the only reasonable way to achieve this seems to be implementing another social idp. 

I have a proof on concept implementation in this fork: https://github.com/JiriHartikka/keycloak/tree/spotify-social-idp. It is just a very simple extension of the AbstractOAuth2IdentityProvider. 

I am willing to write the documentation and test cases for the idp if this is something of interest.

**Comment by pedroigor** [comment]
@JiriHartikka Thanks for the proposal.

We are always looking forward to new integrations. However, we need to run automated tests to make sure the integration do not break.

For that, we have a `SocialLoginTest` test class and you should also updated it with Spotify.

FYI, we have a long-standing PR for Apple ID integration [1]. It is a very popular integration but we could not figure out yet how to have the tests running for it. This is really a blocker, I hope Spotify works differently and we can easily use a test account for running tests.

[1] https://github.com/keycloak/keycloak/pull/7323

**Comment by stianst** [comment]
I don't think using Spotify as a identity provider is a commonly enough wanted capability to warrant adding it out of the box. An alternative would be to deliver it as an extension and register it here https://www.keycloak.org/extensions.html.

**Comment by pedroigor** [comment]
Works for me too. Also aligned with the idea of baking extensions first in
the community before having them in the codebase.

**Reply by thebalaa** [reply]
Anything I can do to help get this pushed through as an extension?

**Reply by pedroigor** [reply]
@thebalaa After you are done with the repository, I think we can just send a PR to our website repository to provide some metadata about your extension.

See https://github.com/keycloak/keycloak-web/tree/main/extensions.
